study.md

```diff
@@ -21,18 +21,49 @@ There is no requirement to follow any links from these pages.
 After doing all the required readings please tell us anything you were confused about.
 
 ```md
-<!-- your answer here -->
+I am confused about nested callbacks and the example given in the article about return:
+
+function foo(){
+    function bar() {
+        return 3;
+    }
+    return bar();
+    function bar() {
+        return 8;
+    }
+}
+alert(foo());
+
+I understand that something about the way bar() is defined both times causes JavaScript to read it before the return but I'm not sure what that thing is.
```

Comment:
Function declarations always get hoisted, so that code snippet is interpreted as:
```javascript
function foo() {
    function bar() {
        return 3;
    }

    function bar() {
        return 8;
    }
    
    return bar();
}

alert(foo());
```